src/Compilers/CSharp/Portable/FlowAnalysis/DefiniteAssignment.cs

```diff
@@ -780,11 +791,15 @@ protected override bool TryGetReceiverAndMember(BoundExpression expr, out BoundE
                         if (Binder.AccessingAutoPropertyFromConstructor(propAccess, this.currentSymbol))
                         {
                             var propSymbol = propAccess.PropertySymbol;
-                            if (propSymbol.IsStatic)
+                            member = (propSymbol as SourcePropertySymbol)?.BackingField;
```

Comment:
Does this leave out any scenarios where the backing field is null, but we still need to check that the receiver was definitely assigned? Perhaps that happens somewhere else?

Reply:
I suspect the check on the receiver happens elsewhere.

We were previously returning `false` in that case but were setting `receiver`, although there is only one caller and that caller ignores `receiver` on `false`.

---
In reply to: [307836981](https://github.com/dotnet/roslyn/pull/37507#discussion_r307836981) [](ancestors = 307836981)